### DISCUSSION TITLE
All linkerd-proxy containers in Linkerd control plane installation return errors

### DISCUSSION THREAD

**Discussion by yegna-08** [linkerd/linkerd2#6415] [Q&A]
I am trying to implement Linkerd stable 2.10.2 in the cluster with Callico CNI , kube version v1.19.12
 
My end goal is to use HAProxy ingress controller with linkerd service mesh

I am new to this so dont know why my linkerd ns control plane pods are in "CrashLoopbackoff" state except the Identity pod

PFA of the pods and logs of them , Please help

All the linkerd-proxy container in the pods face the same error as in the snap and all the applications dont return any logs

![image (1)](https://user-images.githubusercontent.com/86030854/124018452-7f474180-da05-11eb-8e08-59ce64592528.png)
![image](https://user-images.githubusercontent.com/86030854/124018458-80786e80-da05-11eb-8f47-72779b279749.png)

**Comment by cpretzer** [comment]
@yegna-08 the connection refused show that the client is `10.110.3.20`, but that doesn't appear to be any of the pod IP addresses. 

Is it a service address by any chance? Can you get the output from the `linkerd-init` container?

It would also be interesting to see if there are any logs in the calico CNI pods. Are you using the Linkerd CNI plugin?

Did you configure a custom pod CIDR network, by any chance?

**Reply by yegna-08** [reply]
![image](https://user-images.githubusercontent.com/86030854/124061328-4ed6c600-da4c-11eb-9c6a-04b2248976f4.png)

Is it a possible issue ?

**Reply by yegna-08** [reply]
10.110.3.20 is not service IP either.
I tried using both Calico CNI which is installed and also with CNI plugin but same result

**Reply by cpretzer** [reply]
I'd try to figure out what is causing coredns to CrashLoop. I wouldn't be surprised if that is happening even without Linkerd deployed.

Where is this cluster running (GKE, EKS, AKS, bare metal, etc.)?

**Reply by yegna-08** [reply]
bare metal
thank you for addressing

**Reply by yegna-08** [reply]
yes even without Linkerd the core-dns is failing

**Reply by cpretzer** [reply]
Once you resolve the coredns issue, please update this discussion if you have any issues with Linkerd 😄

**Reply by yegna-08** [reply]
sure I am trying to run linkerd in a specific node where core dns is running fine - Can you help on that - I am having trouble using nodeSelector with linkerd install 

I used this 

linkerd install --set nodeSelector:kubernetes.io/hostname=nodename | kubectl apply -f -

but the linkerd proxies are installed in a different node

**Reply by cpretzer** [reply]
If coredns is failing on any of the nodes, then I would expect all the routing to fail between workloads, including Linkerd proxies, that are running on different nodes.

In short, the Linkerd proxy errors are going to be noise until the coredns issue is resolved. The Linkerd proxy errors are the symptom and coredns is the cause, so we need to address the cause first.

**Reply by yegna-08** [reply]
I am working on it , while debugging deleted cluster roles and that started another issue :(

**Reply by yegna-08** [reply]
![image](https://user-images.githubusercontent.com/86030854/124190490-fbae5300-dadf-11eb-9408-0d03c92ccd0e.png)

coredns and calico issues are resolved - now linkerd2 control plane is installed and checks are fine 

But when I tried to install linkerd-viz 

All the pods are running except web - in web the linkerd-proxy logs are attached

**Reply by yegna-08** [reply]
![image](https://user-images.githubusercontent.com/86030854/124190843-8e4ef200-dae0-11eb-8a0c-23b80500a267.png)
